### PR TITLE
core: cable-guy: More than one IP address

### LINKED ISSUE #428: cable-guy: No support for more than one IP on the same interface
As stated by @patrickelectric [here](https://github.com/bluerobotics/companion-docker/pull/412#pullrequestreview-737670558) if there's more than one IP on a given interface, just the first one will be retrieved.

We should support more than one IP per interface, as today in this case the user will incorrectly assume just the first one exists and that the interface is set like that.

DAYS_TO_DEVELOP:4

### PR DESCRIPTION
**Pull request by rafaellehmkuhl**
Before this patch, cable-guy was only able to handle one address per ethernet interface. This was leading to issues like the user only being able to see one IP address on the frontend (even when the interface had multiple), which could cause some misunderstood.

With this patch, the user is now able to:
- See multiple IP addresses associated with each ethernet interface
- Add new IP addresses to a given interface
- Remove specific IP addresses from a given interface

image

Also, some error messages were made more clear, raises were added where necessary and logs from settings are now done with loguru.

Fix #428

### CODE REVIEW

**Comment by rafaellehmkuhl** [comment]
> Is there a docker ?

Yup, same same as the branch

**Review by patrickelectric — CHANGES_REQUESTED** [review]
It's showing three IP blocks, but eth0 here only have 2
image
image

- It's appears possible to set a server DHCP, from the options but nothing happens. 
- When deleting an IP configuration, it still shows for a couple of seconds after finishing the update on the UI. 

When creating a new dynamic IP, more 3 IPs appears as IPv6 here.
image
image

File: core/services/cable_guy/api/manager.py [hunk 1]
```diff
@@ -289,11 +275,61 @@ def set_static_ip(self, interface_name: str, ip: str) -> None:
...
+    def add_ip(self, interface_name: str, mode: AddressMode, ip_address: Optional[str] = None) -> None:
+        """Add IP address to a given interface
+
+        Args:
+            interface_name (str): Interface name
+            ip_address (str): IP address to be added
+        """
+        logger.info(f"Adding IP '{ip_address}' on interface '{interface_name}' in {mode} mode.")
+        try:
+            self.get_interface_by_name(interface_name)
```
[the remark this answers is not in the document]
Comment (rafaellehmkuhl) [reply core/services/cable_guy/api/manager.py:290]: It is, yes. Will add a comment.

**Comment by rafaellehmkuhl** [comment]
Image updated with proper fixes.

**Review by patrickelectric — CHANGES_REQUESTED** [review]
Its failing to change to dynamic IP:
image

image

image

The first link does not change to dynamic or a new dynamic is not added.
If we only support a single dynamic ip, we should remove the dynamic IP option and server option if they are already available.

**Comment by rafaellehmkuhl** [comment]
> Its failing to change to dynamic IP: image
> 
> image
> 
> image
> 
> The first link does not change to dynamic or a new dynamic is not added. If we only support a single dynamic ip, we should remove the dynamic IP option and server option if they are already available.

Yes. I'm aiming with this patch to just support multiples IPs, but not change the manager behavior. I have a patch for the sequence that will do some behavior changes to make it clear to the user what's happening. Basically I changed the DHCP and Dynamic IP creation for a button allowing enabling/disabling DHCP server, as we support only one, and another button to "re-trigger" dynamic IP instead of "adding", as we don't control how many and which IPs the external DHCP server will send.

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**